go/ioutil/meter.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+The meter struct contains time-and-byte-tracking functionality used by
+MeteredReader, MeteredReadCloser, MeteredWriter, MeteredWriteCloser.
+*/
+
+package ioutil
+
+import "time"
+
+// meter contains time-and-byte-tracking functionality.
+type meter struct {
+	fs       []func(b int, d time.Duration)
+	bytes    int64
+	duration time.Duration
+}
+
+// Bytes reports the total bytes read in calls to f so far.
```

Comment:
f ?

Reply:
Hm this refers to the argument `f` in `func measure`, but it's not very helpful here. Let me reword this.